### PR TITLE
(MODULES-4241) Enable Windows acceptance testing

### LINKED ISSUE #197: Add PUPPET_AGENT_STARTUP_MODE=Automatic to the msi installer
By adding PUPPET_AGENT_STARTUP_MODE=Automatic to the msi installer options we can ensure the services that should be started by the msi are actually started.
This is not done if the previous version of the agent was installed with PUPPET_AGENT_STARTUP_MODE=Manual or even Disabled.

### PR DESCRIPTION
**Pull request by MikaelSmith**
Update test setup to work with Windows paths and install behavior.
Fixes #197. Depends on #200.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Still a WIP, but getting pretty close.

**Comment by MikaelSmith** [comment]
Darn, one scenario still fails.

File: lib/facter/puppet_agent_appdata.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Facter.add(:puppet_agent_appdata) do
```
Comment (highb) [inline lib/facter/puppet_agent_appdata.rb]: Oh you should rebase these changes out of this PR.

File: manifests/windows/install.pp [hunk 2]
```diff
@@ -45,6 +45,12 @@
     default => "${::system32}\\cmd.exe"
   }
 
+  if (member($::puppet_agent::service_names, 'puppet')) {
```
Comment (highb) [inline manifests/windows/install.pp]: And these. :)

**Comment by MikaelSmith** [comment]
Just disabled the whole mcollective section, it needs some work. This should now pass.

**Comment by MikaelSmith** [comment]
I ran with
```
bundle exec beaker-hostgenerator redhat7-64ma-windows2012r2-64default.a > hosts.yaml
export BEAKER_setfile=hosts.yaml
export PUPPET_INSTALL_TYPE=agent
bundle exec rake beaker
```

**Comment by highb** [comment]
👍  lgtm

**Pull request merged**